File: mobile/android/SyncProxyAndroid/src/com/ford/syncV4/proxy/rpc/PerformAudioPassThru.java
package com.ford.syncV4.proxy.rpc;

import java.util.Hashtable;
import java.util.Vector;

import com.ford.syncV4.proxy.RPCRequest;
import com.ford.syncV4.proxy.constants.Names;
import com.ford.syncV4.proxy.rpc.enums.BitsPerSample;
import com.ford.syncV4.proxy.rpc.enums.AudioType;
import com.ford.syncV4.proxy.rpc.enums.SamplingRate;
import com.ford.syncV4.util.DebugTool;

public class PerformAudioPassThru extends RPCRequest {

    public PerformAudioPassThru() {
        super("PerformAudioPassThru");
    }
    public PerformAudioPassThru(Hashtable hash) {
        super(hash);
    }
    public void setInitialPrompt(Vector<TTSChunk> initialPrompt) {
    	if (initialPrompt != null) {
    		parameters.put(Names.initialPrompt, initialPrompt);
    	} else {
    		parameters.remove(Names.initialPrompt);
    	}
    }
    public Vector<TTSChunk> getInitialPrompt() {
    	if (parameters.get(Names.initialPrompt) instanceof Vector<?>) {
	        Vector<?> list = (Vector<?>)parameters.get(Names.initialPrompt);
	        if (list != null && list.size() > 0) {
	            Object obj = list.get(0);
	            if (obj instanceof TTSChunk) {
	                return (Vector<TTSChunk>) list;
	            } else if (obj instanceof Hashtable) {
	                Vector<TTSChunk> newList = new Vector<TTSChunk>();
	                for (Object hashObj : list) {
	                    newList.add(new TTSChunk((Hashtable)hashObj));
	                }
	                return newList;
	            }
	        }
    	}
        return null;
    }
    public void setAudioPassThruDisplayText1(String audioPassThruDisplayText1) {
    	if (audioPassThruDisplayText1 != null) {
    		parameters.put(Names.audioPassThruDisplayText1, audioPassThruDisplayText1);
    	} else {
    		parameters.remove(Names.audioPassThruDisplayText1);
    	}
    }
    public String getAudioPassThruDisplayText1() {
    	return (String) parameters.get(Names.audioPassThruDisplayText1);
    }
    public void setAudioPassThruDisplayText2(String audioPassThruDisplayText2) {
    	if (audioPassThruDisplayText2 != null) {
    		parameters.put(Names.audioPassThruDisplayText2, audioPassThruDisplayText2);
    	} else {
    		parameters.remove(Names.audioPassThruDisplayText2);
    	}
    }
    public String getAudioPassThruDisplayText2() {
    	return (String) parameters.get(Names.audioPassThruDisplayText2);
    }
    public void setSamplingRate(SamplingRate samplingRate) {
    	if (samplingRate != null) {
    		parameters.put(Names.samplingRate, samplingRate);
    	} else {
    		parameters.remove(Names.samplingRate);
    	}
    }
    public SamplingRate getSamplingRate() {
    	Object obj = parameters.get(Names.samplingRate);
    	if (obj instanceof SamplingRate) {
    		return (SamplingRate) obj;
    	} else if (obj instanceof String) {
    		SamplingRate theCode = null;
            try {
                theCode = SamplingRate.valueForString((String) obj);
            } catch (Exception e) {
            	DebugTool.logError("Failed to parse " + getClass().getSimpleName() + "." + Names.samplingRate, e);
            }
            return theCode;
    	}
        return null;
    }
    public void setMaxDuration(Integer maxDuration) {
    	if (maxDuration != null) {
    		parameters.put(Names.maxDuration, maxDuration);
    	} else {
    		parameters.remove(Names.maxDuration);
    	}
    }
    public int getMaxDuration() {
    	return (Integer) parameters.get(Names.maxDuration);
    }
    public void setBitsPerSample(BitsPerSample audioQuality) {
    	if (audioQuality != null) {
    		parameters.put(Names.bitsPerSample, audioQuality);
    	} else {
    		parameters.remove(Names.bitsPerSample);
    	}
    }
    public BitsPerSample getBitsPerSample() {
    	Object obj = parameters.get(Names.bitsPerSample);
    	if (obj instanceof BitsPerSample) {
    		return (BitsPerSample) obj;
    	} else if (obj instanceof String) {
    		BitsPerSample theCode = null;
            try {
                theCode = BitsPerSample.valueForString((String) obj);
            } catch (Exception e) {
            	DebugTool.logError("Failed to parse " + getClass().getSimpleName() + "." + Names.bitsPerSample, e);
            }
            return theCode;
    	}
        return null;
    }
    public void setAudioType(AudioType audioType) {
    	if (audioType != null) {
    		parameters.put(Names.audioType, audioType);
    	} else {
    		parameters.remove(Names.audioType);
    	}
    }
    public AudioType getAudioType() {
    	Object obj = parameters.get(Names.audioType);
    	if (obj instanceof AudioType) {
    		return (AudioType) obj;
    	} else if (obj instanceof String) {
    		AudioType theCode = null;
            try {
                theCode = AudioType.valueForString((String) obj);
            } catch (Exception e) {
            	DebugTool.logError("Failed to parse " + getClass().getSimpleName() + "." + Names.audioType, e);
            }
            return theCode;
    	}
        return null;
    }
}

File: src/components/JSONHandler/src/RPC2ObjectsImpl/NsRPC2Communication/UI/PerformAudioPassThru.cpp
//
// Copyright (c) 2013, Ford Motor Company
// All rights reserved.
//
// Redistribution and use in source and binary forms, with or without
// modification, are permitted provided that the following conditions are met:
//
// Redistributions of source code must retain the above copyright notice, this
// list of conditions and the following disclaimer.
//
// Redistributions in binary form must reproduce the above copyright notice,
// this list of conditions and the following
// disclaimer in the documentation and/or other materials provided with the
// distribution.
//
// Neither the name of the Ford Motor Company nor the names of its contributors
// may be used to endorse or promote products derived from this software
// without specific prior written permission.
//
// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
// POSSIBILITY OF SUCH DAMAGE.
//

#include "../src/../include/JSONHandler/RPC2Objects/NsRPC2Communication/UI/PerformAudioPassThru.h"
#include "../src/../include/JSONHandler/RPC2Objects/Marshaller.h"

/*
  interface	NsRPC2Communication::UI
  version	1.2
  generated at	Thu Jan 24 06:41:15 2013
  source stamp	Wed Jan 23 13:56:28 2013
  author	RC
*/

using namespace NsRPC2Communication::UI;


PerformAudioPassThru& PerformAudioPassThru::operator =(const PerformAudioPassThru& c)
{
  initialPrompt=c.initialPrompt;
  if(audioPassThruDisplayText1)  delete audioPassThruDisplayText1;
  audioPassThruDisplayText1= c.audioPassThruDisplayText1 ? new std::string(c.audioPassThruDisplayText1[0]) : 0;
  if(audioPassThruDisplayText2)  delete audioPassThruDisplayText2;
  audioPassThruDisplayText2= c.audioPassThruDisplayText2 ? new std::string(c.audioPassThruDisplayText2[0]) : 0;
  samplingRate=c.samplingRate;
  maxDuration=c.maxDuration;
  bitsPerSample=c.bitsPerSample;
  audioType=c.audioType;
  appId=c.appId;
  return *this;
}


PerformAudioPassThru::~PerformAudioPassThru(void)
{
  if(audioPassThruDisplayText1)  delete audioPassThruDisplayText1;
  if(audioPassThruDisplayText2)  delete audioPassThruDisplayText2;
}


PerformAudioPassThru::PerformAudioPassThru(void) : 
  RPC2Request(Marshaller::METHOD_NSRPC2COMMUNICATION_UI__PERFORMAUDIOPASSTHRU),
  audioPassThruDisplayText1(0),
  audioPassThruDisplayText2(0)
{
}


PerformAudioPassThru::PerformAudioPassThru(const PerformAudioPassThru& c) : RPC2Request(Marshaller::METHOD_NSRPC2COMMUNICATION_UI__PERFORMAUDIOPASSTHRU,c.getId())
{
  *this=c;
}


const std::vector< NsSmartDeviceLinkRPCV2::TTSChunk>& PerformAudioPassThru::get_initialPrompt(void)
{
  return initialPrompt;
}

bool PerformAudioPassThru::set_initialPrompt(const std::vector< NsSmartDeviceLinkRPCV2::TTSChunk>& initialPrompt_)
{
  initialPrompt=initialPrompt_;
  return true;
}

const std::string* PerformAudioPassThru::get_audioPassThruDisplayText1(void)
{
  return audioPassThruDisplayText1;
}

bool PerformAudioPassThru::set_audioPassThruDisplayText1(const std::string& audioPassThruDisplayText1_)
{
  if(audioPassThruDisplayText1)  delete audioPassThruDisplayText1;
  audioPassThruDisplayText1=new std::string(audioPassThruDisplayText1_);
  return true;
}

void PerformAudioPassThru::reset_audioPassThruDisplayText1(void)
{
  if(audioPassThruDisplayText1)  delete audioPassThruDisplayText1;
  audioPassThruDisplayText1=0;
}

const std::string* PerformAudioPassThru::get_audioPassThruDisplayText2(void)
{
  return audioPassThruDisplayText2;
}

bool PerformAudioPassThru::set_audioPassThruDisplayText2(const std::string& audioPassThruDisplayText2_)
{
  if(audioPassThruDisplayText2)  delete audioPassThruDisplayText2;
  audioPassThruDisplayText2=new std::string(audioPassThruDisplayText2_);
  return true;
}

void PerformAudioPassThru::reset_audioPassThruDisplayText2(void)
{
  if(audioPassThruDisplayText2)  delete audioPassThruDisplayText2;
  audioPassThruDisplayText2=0;
}

const NsSmartDeviceLinkRPCV2::SamplingRate& PerformAudioPassThru::get_samplingRate(void)
{
  return samplingRate;
}

bool PerformAudioPassThru::set_samplingRate(const NsSmartDeviceLinkRPCV2::SamplingRate& samplingRate_)
{
  samplingRate=samplingRate_;
  return true;
}

unsigned int PerformAudioPassThru::get_maxDuration(void)
{
  return maxDuration;
}

bool PerformAudioPassThru::set_maxDuration(unsigned int maxDuration_)
{
  maxDuration=maxDuration_;
  return true;
}

const NsSmartDeviceLinkRPCV2::AudioCaptureQuality& PerformAudioPassThru::get_bitsPerSample(void)
{
  return bitsPerSample;
}

bool PerformAudioPassThru::set_bitsPerSample(const NsSmartDeviceLinkRPCV2::AudioCaptureQuality& bitsPerSample_)
{
  bitsPerSample=bitsPerSample_;
  return true;
}

const NsSmartDeviceLinkRPCV2::AudioType& PerformAudioPassThru::get_audioType(void)
{
  return audioType;
}

bool PerformAudioPassThru::set_audioType(const NsSmartDeviceLinkRPCV2::AudioType& audioType_)
{
  audioType=audioType_;
  return true;
}

int PerformAudioPassThru::get_appId(void)
{
  return appId;
}

bool PerformAudioPassThru::set_appId(int appId_)
{
  appId=appId_;
  return true;
}

bool PerformAudioPassThru::checkIntegrity(void)
{
  return PerformAudioPassThruMarshaller::checkIntegrity(*this);
}
